src/devicecontroller/DefaultDeviceController.ts

```diff
@@ -454,6 +462,26 @@ export default class DefaultDeviceController implements DeviceControllerBasedMed
     this.alreadyHandlingDeviceChange = false;
   }
 
+  private async handleDeviceStreamEnded(kind: string, deviceId: string): Promise<void> {
+    if (
```

Comment:
I don't think this `if` statement is required. Calling `chooseInputDevice(video, null, false)` below will eventually stop local video tile.

Stack trace
1. `chooseInputDevice(video, null, false)`
2. `this.releaseMediaStream(this.activeDevices[kind].stream)`
3. In `releaseMediaStream`, you call stop `localVideoTile()` of matching active devices

Reply:
Yes, I confirmed it because I could never trigger that code to call if we put it after chooseInputDevice(null) but it depends on the value of this.activeDevices[kind].stream so I chose to keep it just in case. Let me know if you think it is totally unnecessary and I will remove it.

Reply:
Removed since handleDeviceStreamEnded only called when activeDevices[kind].stream exists so stopLocalVideoTile should always be called.